### PR TITLE
Added ViTSTR small model

### PR DESCRIPTION
**Pull request by Anna-Mironova**

### CODE REVIEW

**Comment by eaidova** [comment]
TO DO:

- [ ] update demo to support model

File: models/public/vitstr-small-patch16-224/README.md [hunk 1]
```diff
@@ -0,0 +1,98 @@
+# vitstr-small-patch16-224
+
+## Use-case and high-level description
+
+The `vitstr-small-patch16-224` model is `small` version of the ViTSTR models.  ViTSTR is a simple single-stage model that uses a pre-trained Vision Transformer (ViT) to perform Scene Text Recognition (ViTSTR).  Small version of model requires an embedding size of 384 and number of heads of 6. Model is able to recognize alphanumeric text.
```
Comment (eaidova) [inline models/public/vitstr-small-patch16-224/README.md:5]: I believe that model implementation requires this architecture, but we work with already designed model so
```suggestion
The `vitstr-small-patch16-224` model is `small` version of the ViTSTR models.  ViTSTR is a simple single-stage model that uses a pre-trained Vision Transformer (ViT) to perform Scene Text Recognition (ViTSTR).  Small version of model has an embedding size of 384 and number of heads of 6. Model is able to recognize alphanumeric text.
```

Comment (eaidova) [reply models/public/vitstr-small-patch16-224/README.md:5]: and BTW, alphanumeric text means only alphabet and numbers, but as I can see from described symbols below, it also support additional symbols

File: models/public/vitstr-small-patch16-224/README.md [hunk 2]
```diff
@@ -0,0 +1,98 @@
...
+## Output
+
+### Original model
+
+Output tensor, name: `logits`, shape: `1, 25, 96` in the format `B, W, L`, where:
+
+- `B` - batch size
+- `W` - output sequence length
+- `L` - confidence distribution across alphanumeric symbols:
+  `[GO][s]0123456789abcdefghijklmnopqrstuvwxyzABCDEFGHIJKLMNOPQRSTUVWXYZ!"#$%&'()*+,-./:;<=>?@[\]^_`{|}~`, where [s] - special end of sequence character for decoder, [GO] - special start token for decoder.
```
Comment (eaidova) [inline models/public/vitstr-small-patch16-224/README.md:65]: very-very long line,
maybe better describe supported symblos in the file?

File: demos/tests/cases.py [hunk 3]
```diff
@@ -523,6 +523,11 @@ def single_option_cases(key, *args):
                              extra_models=[ModelArg('text-recognition-0015-decoder')]),
                     TestCase(options={'-m_tr': ModelArg('text-recognition-resnet-fc'),
                                       '-tr_pt_first': None}),
+                    TestCase(options={'-m_tr': ModelArg('vitstr-small-patch16-224'),
+                                      '-tr_pt_first': None,
+                                      '-m_tr_ss': '0123456789abcdefghijklmnopqrstuvwxyzABCDEFGHIJKLMNOPQRSTUVWXYZ!"#$%&\'()*+,-./:;<=>?@[\\]^_`{|}~',
```
Comment (eizamaliev) [inline demos/tests/cases.py:528]: I recommend add support for `vocab.txt` loading instead of explicit set.

File: demos/text_detection_demo/cpp/README.md [hunk 4]
```diff
@@ -12,6 +12,10 @@ The demo shows an example of using neural networks to detect and recognize print
 * `text-recognition-0015`, which is a recognition network for recognizing text. You should add options `-tr_pt_first`, `-m_tr_ss "?0123456789abcdefghijklmnopqrstuvwxyzABCDEFGHIJKLMNOPQRSTUVWXYZ"` (supported symbols set), `-tr_o_blb_nm "logits"` (to specify output name) and `-dt simple` (to specify decoder type). You can also specify `-lower` option to convert predicted text to lower-case. See model [description](../../../models/intel/text-recognition-0015/README.md) for details.
 * `text-recognition-resnet-fc`, which is a recognition network for recognizing text. You should add option `-tr_pt_first`.
 * `handwritten-score-recognition-0001`, which is a recognition network for recognizing handwritten score marks like `<digit>` or `<digit>.<digit>`.
+* `vitstr-small-patch16-224`, which is a recognition network for recognizing text. You should add options:
+    ```
+    -tr_pt_first, -m_tr_ss 0123456789abcdefghijklmnopqrstuvwxyzABCDEFGHIJKLMNOPQRSTUVWXYZ!"#$%&'()*+,-./:;<=>?@[\]^_`{|}~ (supported symbols set), -dt simple (to specify decoder type), -start_index 1 (to process output from provided index) and -pad " " (to use specific pad symbol)
```
Comment (eizamaliev) [inline demos/text_detection_demo/cpp/README.md:17]: It seems, that this line contains both CL arguments and comments for them. In that case it uncopypastable.

File: models/public/vitstr-small-patch16-224/accuracy-check.yml [hunk 5]
```diff
@@ -0,0 +1,303 @@
+models:
+  - name: vitstr-small-patch16-224-onnx
+    launchers:
+      - framework: onnx_runtime
+        model: vitstr-small-patch16-224.onnx
+        adapter:
+          type: simple_decoder
+          start_index: 1
+          do_lower: True
+          custom_label_map:
```
Comment (eaidova) [inline models/public/vitstr-small-patch16-224/accuracy-check.yml:10]: could you please implement vocab.txt based initialization for label map in adapter?

File: demos/text_detection_demo/cpp/main.cpp [hunk 6]
```diff
@@ -119,10 +119,13 @@ int main(int argc, char *argv[]) {
         auto link_conf_threshold = static_cast<float>(FLAGS_link_pixel_thr);
         auto decoder_type = FLAGS_dt;
         auto decoder_bandwidth = FLAGS_b;
+        auto decoder_start_index = FLAGS_start_index;
+        auto pad = FLAGS_pad;
+
+        const char kPadSymbol = pad[0];
```
Comment (Wovchena) [inline demos/text_detection_demo/cpp/main.cpp:125]: Please throw if `pad.size()` is not 1.

File: demos/text_detection_demo/cpp/main.cpp [hunk 7]
```diff
@@ -119,16 +119,38 @@ int main(int argc, char *argv[]) {
...
+        auto pad = FLAGS_pad;
+        auto vocab_file = FLAGS_vocab_tr;
+
+        if (pad.length() != 1)
+            throw std::invalid_argument("Pad symbol should be 1 character");
+
+        const char kPadSymbol = pad[0];
+
+        auto m_tr_ss = FLAGS_m_tr_ss;
+        if (!FLAGS_vocab_tr.empty()) {
```
[the remark this answers is not in the document]
Comment (Anna-Mironova) [reply demos/text_detection_demo/cpp/main.cpp]: Done

File: demos/text_detection_demo/cpp/main.cpp [hunk 8]
```diff
@@ -119,16 +119,38 @@ int main(int argc, char *argv[]) {
...
+            std::string symbols = "";
+            std::ifstream inputFile(FLAGS_vocab_tr);
+            if (!inputFile.is_open())
+                throw std::runtime_error("Can't open the vocab file: " + FLAGS_vocab_tr);
+            std::string vocabLine;
+            while (std::getline(inputFile, vocabLine)) {
+                symbols += vocabLine;
```
[the remark this answers is not in the document]
Comment (Anna-Mironova) [reply demos/text_detection_demo/cpp/main.cpp:145]: Done

**Review by Wovchena — APPROVED** [review]

**Review by eizamaliev — APPROVED** [review]

**Comment by eaidova** [comment]
Jenkins please retry a build

File: tools/accuracy_checker/accuracy_checker/adapters/text_recognition.py [hunk 9]
```diff
@@ -260,6 +262,15 @@ def configure(self):
         if self.custom_label_map:
             labels = {int(k): v for k, v in self.custom_label_map.items()}
             self.custom_label_map = labels
+        vocab_file = self.get_value_from_config("vocabulary_file")
+        if not self.custom_label_map and vocab_file:
+            chr_list = []
+            start_label = self.get_value_from_config('start_label')
+            with open(vocab_file, 'rb') as vocab:
```
Comment (eaidova) [inline tools/accuracy_checker/accuracy_checker/adapters/text_recognition.py:269]: why do you read text fil as binary?

Comment (Anna-Mironova) [reply tools/accuracy_checker/accuracy_checker/adapters/text_recognition.py]: Done

**Review by eaidova — APPROVED** [review]
